packages/plugin-core/src/features/TipOfTheDayWebview.ts

```diff
@@ -0,0 +1,198 @@
+import { VSCodeEvents } from "@dendronhq/common-all";
+import { MetadataService } from "@dendronhq/engine-server";
+import _md from "markdown-it";
+import * as vscode from "vscode";
+import {
+  DisplayLocation,
+  FeatureShowcaseUserResponse,
+  IFeatureShowcaseMessage,
+} from "../showcase/IFeatureShowcaseMessage";
+import { AnalyticsUtils } from "../utils/analytics";
+
+/**
+ * Side Panel webview that shows the tip of the day
+ * TODO: Add functionality
+ *  - let user rotate tips
+ */
+export default class TipOfTheDayWebview implements vscode.WebviewViewProvider {
+  private _webview: vscode.WebviewView | undefined;
+  private _tips: IFeatureShowcaseMessage[];
+  private _curTipIndex;
+
+  private BUTTON_CLICKED_MSG = "buttonClicked";
+  private TIP_SHOWN_MSG = "loaded";
+
+  private get _currentTip(): IFeatureShowcaseMessage {
+    return this._tips[this._curTipIndex];
+  }
+
+  /**
+   * The set of tips to show the user.
+   * @param tips
+   */
+  constructor(tips: IFeatureShowcaseMessage[]) {
+    this._tips = tips;
+
+    // Get the last seen tip from storage. If they've never seen one before,
+    // then set them at a random index so we can get an even distribution of
+    // tips shown across the population.
+    let storedIndex = MetadataService.instance().TipOfDayIndex;
+
+    if (!storedIndex) {
+      storedIndex = Math.random() * this._tips.length;
+    }
+
+    this._curTipIndex = storedIndex;
+  }
+
+  resolveWebviewView(
+    webviewView: vscode.WebviewView,
+    _context: vscode.WebviewViewResolveContext<unknown>,
+    _token: vscode.CancellationToken
+  ): void {
+    this._webview = webviewView;
+
+    webviewView.webview.options = {
+      enableScripts: true,
+      enableCommandUris: true,
+    };
+
+    webviewView.webview.html = this.getContent(this._currentTip);
+
+    webviewView.webview.onDidReceiveMessage(
+      async (message) => {
+        switch (message.command) {
+          case this.TIP_SHOWN_MSG:
+            AnalyticsUtils.track(VSCodeEvents.FeatureShowcaseDisplayed, {
+              messageType: message.showcaseEntry,
+              displayLocation: DisplayLocation.TipOfTheDayView,
+            });
+            return;
+
+          case this.BUTTON_CLICKED_MSG: {
+            AnalyticsUtils.track(VSCodeEvents.FeatureShowcaseResponded, {
+              messageType: message.showcaseEntry,
+              displayLocation: DisplayLocation.TipOfTheDayView,
+              userResponse: FeatureShowcaseUserResponse.confirmed,
+            });
+
+            if (this._currentTip.onConfirm) {
+              const fn = this._currentTip.onConfirm.bind(this._currentTip);
+              fn();
+            }
+
+            return;
+          }
+          default:
+            break;
+        }
+      },
+      undefined,
+      undefined
+    );
+
+    this.showNextTip();
+  }
+
+  private showNextTip(): boolean {
+    this._curTipIndex = (this._curTipIndex + 1) % this._tips.length;
```

Comment:
Tip of the day resulted in an error for me: `An error occurred while loading view: dendron.tip-of-the-day`. 
It might be because we get a floating-point number from this and we a undefined tip is returned from [here](https://github.com/dendronhq/dendron/pull/2832/files#diff-6491d89c0e1ed956f06220885819e59144ad233fc28be7afa8702362ab50795bR26)